examples/linear_model/plot_poisson_regression_non_normal_loss.py

```diff
@@ -237,22 +254,21 @@ def score_estimator(estimator, df_test):
     ],
     remainder="drop",
 )
-rf = make_pipeline(
-    rf_preprocessor,
-    RandomForestRegressor(min_weight_fraction_leaf=0.01, n_jobs=2)
-)
-rf.fit(df_train, df_train["Frequency"].values,
-       randomforestregressor__sample_weight=df_train["Exposure"].values)
-
+gbrt = Pipeline([
+    ("preprocessor", tree_preprocessor),
+    ("regressor", HistGradientBoostingRegressor(max_leaf_nodes=128)),
```

Comment:
With `min_weight_fraction_leaf=0.01`, we could avoid non-positive predictions of the random forest. Could we try the same for the HGBR, in this case with `min_samples_leaf=...`?

Reply:
Because `max_leaf_nodes` is limited and the number of trees very large, I don't think this is going to change anything. Also with the Poisson loss and its log link that are now enabled we don't have any negative predictions for HGBRT anymore.